core/src/main/java/org/apache/carbondata/core/keygenerator/directdictionary/timestamp/TimeStampDirectDictionaryGenerator.java

```diff
@@ -117,9 +113,11 @@ private TimeStampDirectDictionaryGenerator() {
    * @return dictionary value
    */
   @Override public int generateDirectSurrogateKey(String memberStr) {
-    SimpleDateFormat timeParser = new SimpleDateFormat(CarbonProperties.getInstance()
-        .getProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT,
-            CarbonCommonConstants.CARBON_TIMESTAMP_DEFAULT_FORMAT));
+    SimpleDateFormat timeParser = threadLocal.get();
+    if(timeParser == null){
+      timeParser = new SimpleDateFormat(dateFormat);
+      threadLocal.set(timeParser);
+    }
     timeParser.setLenient(false);
```

Comment:
Please extract above codes to a new initial method,  and invoke this method in different thread.
It it not good to run these codes in generateDirectSurrogateKey method.